src/adapter/src/coord/sequencer.rs

```diff
@@ -3631,6 +3646,25 @@ impl<S: Append + 'static> Coordinator<S> {
         }
     }
 
+    fn update_metrics_retention(&mut self) {
+        let duration = self.catalog.system_config().metrics_retention();
+        let policy = ReadPolicy::lag_writes_by(Timestamp::new(
+            u64::try_from(duration.as_millis()).unwrap_or_else(|_e| {
+                tracing::error!("Absurd metrics retention duration: {duration:?}.");
+                u64::MAX
+            }),
+        ));
+        let ids = self
+            .catalog
+            .entries()
+            .filter(|entry| entry.item().is_retained_metrics_relation())
+            .map(|entry| entry.id())
+            .collect::<Vec<_>>();
+        for id in ids {
+            self.update_storage_base_read_policy(id, policy.clone())
+        }
```

Comment:
We should convert `update_storage_base_read_policy` to take a `Vec` of IDs and policies and only send one message to storage instead of sending one message per id.